pkg/controller/backupstoragelocation_controller.go

```diff
@@ -57,36 +70,50 @@ func (r *BackupStorageLocationReconciler) Reconcile(req ctrl.Request) (ctrl.Resu
 		location := &locationList.Items[i]
 		log := r.Log.WithField("controller", "backupstoragelocation").WithField("backupstoragelocation", location.Name)
 
-		if location.Name == r.StorageLocation.DefaultStorageLocation {
+		if location.Name == r.DefaultBackupLocationInfo.StorageLocation {
 			defaultFound = true
 		}
 
-		if !r.StorageLocation.IsReadyToValidate(location, log) {
+		pluginManager := r.NewPluginManager(log)
+		defer pluginManager.CleanupClients()
```

Comment:
can this be done outside the `for` loop?

Reply:
Hmmm yeah, good point!

Reply:
👍